### ISSUE TITLE
Is storing state data inside of a grain redundant?

### ISSUE THREAD

**Issue by eduherminio** [dotnet/orleans#5316]
Hi guys.
I'm wondering if it makes any sense to store State data inside a grain.
I'm adding State to a `Foo` grain.

When it was stateless (`Foo : Grain`), having `_bar` did make sense.
Now that a state has been added and persisted, Foo's Bar can be accessed though its State, so I totally see `_bar` as redundant info, but just in case I'm missing something.

```c#
    [StorageProvider(ProviderName = "MyProvider")]
    public class Foo : Grain<Bar>
    {
        private readonly Bar _bar;  // Redundant?
        // ...
    }

    public class Bar
    {
        public int A { get; set; }

        public double B { get; set; }
    }
```

Thanks in advance for your answers!

**Comment by sergeybykov** [comment]
Yes, if for `Foo : Grain<Bar>` it would be redundant to keep another copy of the same state of type `Bar` in `_bar`.
